File: src/main/java/com/somefrills/features/tweaks/BreakResetFix.java
package com.somefrills.features.tweaks;

import com.somefrills.config.FrillsConfig;
import com.somefrills.events.InventoryUpdateEvent;
import com.somefrills.features.core.Feature;
import com.somefrills.features.core.FrillsFeature;
import com.somefrills.mixin.ClientPlayerInteractionManagerAccessor;
import meteordevelopment.orbit.EventHandler;

import static com.somefrills.Main.mc;

@FrillsFeature
public class BreakResetFix extends Feature {

    public BreakResetFix() {
        super(FrillsConfig.tweaks.breakResetFixEnabled);
    }
    @EventHandler
    public void onBreakReset(InventoryUpdateEvent event) {
        if (mc == null || mc.player == null || mc.interactionManager == null) return;
        if (event.slotId >= 36 && event.slotId <= 44 && mc.player.getInventory().getSelectedSlot() == event.slotId - 36) {
            ((ClientPlayerInteractionManagerAccessor) mc.interactionManager).setStack(event.stack);
        } // manually update the variable once the server updates our held item, prevents the mismatch and thus fixes the break cancel
    }
}

File: src/main/kotlin/com/somefrills/features/tweaks/BreakResetFix.kt
// TODO(Ravel): Failed to fully resolve file: null cannot be cast to non-null type com.intellij.psi.PsiClass
package com.somefrills.features.tweaks

import com.somefrills.Main.mc
import com.somefrills.config.FrillsMod

import com.somefrills.events.InventoryUpdateEvent
import com.somefrills.features.core.Feature
import com.somefrills.modules.FrillsFeature
import com.somefrills.mixin.MultiPlayerGameModeAccessor
import com.somefrills.events.core.EventHandle

@FrillsFeature
object BreakResetFix : Feature(FrillsMod.config.tweaks.breakResetFixEnabled) {
    @EventHandle
    fun onBreakReset(event: InventoryUpdateEvent) {
        if (mc.player == null || mc.gameMode == null) return
        if (event.slotId in 36..44 && mc.player?.getInventory()
                ?.selectedSlot == event.slotId - 36
        ) {
            (mc.gameMode as MultiPlayerGameModeAccessor).setStack(event.stack)
        } // manually update the variable once the server updates our held item, prevents the mismatch and thus fixes the break cancel
    }
}
